src/installer/records.py

```diff
@@ -0,0 +1,73 @@
+__all__ = [
+    "Hash",
+    "Record",
+    "SuperfulousRecordColumnsWarning",
+    "parse_record_file",
+]
+
+import csv
+import warnings
+
+from installer._compat import pathlib
+from installer._compat.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Iterator, Optional
+
+
+class SuperfulousRecordColumnsWarning(UserWarning):
+    pass
+
+
+class Hash(object):
+    def __init__(self, name, value):
+        # type: (str, str) -> None
+        self.name = name
+        self.value = value
+
+    def __repr__(self):
+        return "Hash(name={!r}, value={!r})".format(self.name, self.value)
+
+    @classmethod
+    def parse(cls, h):
+        # type: (str) -> Hash
+        name, value = h.split("=", 1)
+        return Hash(name, value)
+
+
+class Record(object):
+    def __init__(self, path, hash_, size):
+        # type: (pathlib.PurePosixPath, Optional[Hash], Optional[int]) -> None
+        self.path = path
+        self.hash_ = hash_
+        self.size = size
+
+    def __repr__(self):
+        # type: () -> str
+        return "Record(path={!r}, hash_={!r}, size={!r})".format(
+            self.path, self.hash_, self.size,
+        )
+
+    @classmethod
+    def parse(cls, p, h, s):
+        # type: (str, str, str) -> Record
+        return cls(
+            path=pathlib.PurePosixPath(p),
+            hash_=Hash.parse(h) if h else None,
+            size=int(s) if s else None,
+        )
+
+
+def parse_record_file(f):
+    # type: (Iterator[str]) -> Iterator[Record]
+    for i, row in enumerate(csv.reader(f)):
```

Comment:
nitpick: I usually name these position arguments ``at_pos`` as feel makes the code easier to read in the body, ``i`` doesn't tell much what it encodes; generally reserve single character variables for comprehension 🤔